yarn-project/circuit-types/src/tx/tx_hash.ts

```diff
@@ -1,32 +1,68 @@
 import { Fr } from '@aztec/circuits.js';
-import { Buffer32 } from '@aztec/foundation/buffer';
 import { schemas } from '@aztec/foundation/schemas';
+import { BufferReader } from '@aztec/foundation/serialize';
+
+import { z } from 'zod';
 
 /**
  * A class representing hash of Aztec transaction.
  */
-export class TxHash extends Buffer32 {
+export class TxHash {
   constructor(
-    /** The buffer containing the hash. */
-    hash: Buffer,
-  ) {
-    super(hash);
-  }
+    /** A field representing the tx hash (tx hash is an output of poseidon hash hence it's a field). */
+    public readonly hash: Fr,
+  ) {}
 
   /*
    * TxHashes are generated from the first nullifier of a transaction, which is a Fr.
-   * Using Buffer32.random() could potentially generate invalid TxHashes.
    * @returns A random TxHash.
    */
-  static override random() {
-    return new TxHash(Fr.random().toBuffer());
+  static random() {
+    return new TxHash(Fr.random());
+  }
+
+  static fromBuffer(buffer: Uint8Array | BufferReader) {
+    const reader = BufferReader.asReader(buffer);
+    return new this(reader.readObject(Fr));
+  }
+
+  static fromString(str: string) {
+    return new TxHash(Fr.fromString(str));
+  }
+
+  static fromBigInt(value: bigint) {
+    return new TxHash(new Fr(value));
+  }
+
+  public toBuffer() {
+    return this.hash.toBuffer();
+  }
+
+  public toString() {
+    return this.hash.toString();
+  }
+
+  public toBigInt() {
+    return this.hash.toBigInt();
+  }
+
+  public equals(other: TxHash) {
+    return this.hash.equals(other.hash);
   }
 
   static get schema() {
-    return schemas.BufferHex.transform(value => new TxHash(value));
+    return z
+      .object({
+        hash: schemas.Fr,
+      })
+      .transform(({ hash }) => new TxHash(hash));
```

Comment:
Curious why this change. Before, a `TxHash` would be represented as just a string like `0x1234`, but with this change it'd be an object like `{ "hash": "0x1234" }`. Personally I'd prefer the more compact representation.

Reply:
I just thought that's what I need to do when doing the change to Fr (didn't really understand the schemas when first fiddling with it). I see that it was unnecessary so I reverted it in https://github.com/AztecProtocol/aztec-packages/pull/10954/commits/2c021b7851f067bb506577e9e7a38e2479398434

Thanks for the feedback and the quick review